src/Features/CSharp/Portable/Structure/CSharpBlockStructureProvider.cs

```diff
@@ -39,6 +39,7 @@ internal class CSharpBlockStructureProvider : AbstractBlockStructureProvider
             builder.Add<OperatorDeclarationSyntax, OperatorDeclarationStructureProvider>();
             builder.Add<ParenthesizedLambdaExpressionSyntax, ParenthesizedLambdaExpressionStructureProvider>();
             builder.Add<PropertyDeclarationSyntax, PropertyDeclarationStructureProvider>();
+            builder.Add<RecordDeclarationSyntax, TypeDeclarationStructureProvider>();
```

Comment:
Out of curiosity, what does this line (and class) do?

Reply:
So the way 'outlining' works is that we have a tpo level walker that descends through teh tree looking to see how they should be outlined.  The mapping here maps between the concrete type of the node in teh tree to the "outlining provider" that will outline it.  In this case, it's saying "hey, if i hit a REcordDeclarationSyntax, then we should outline it using the TYpeDeclStructureProvider". :)